pkg/pod/status_test.go

```diff
@@ -190,6 +195,391 @@ func TestSetTaskRunStatusBasedOnStepStatus_sidecar_logs(t *testing.T) {
+				CompletionTime: &metav1.Time{Time: time.Now()},
+			},
+		},
+	}, {
+		desc:                 "test result (with task run result defined) with no result without signed termination message",
+		specifyTaskRunResult: true,
+		podStatus: corev1.PodStatus{
+			Phase: corev1.PodSucceeded,
+			ContainerStatuses: []corev1.ContainerStatus{{
+				Name: "step-bar",
+				State: corev1.ContainerState{
+					Terminated: &corev1.ContainerStateTerminated{
+						Message: "[]",
+					},
+				},
+			}},
+		},
+		want: v1beta1.TaskRunStatus{
+			Status: statusSRVUnverified(),
+			TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+				Steps: []v1beta1.StepState{{
+					ContainerState: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{
+							Message: "[]",
+						}},
+					Name:          "bar",
+					ContainerName: "step-bar",
+				}},
+				Sidecars: []v1beta1.SidecarState{},
+				// We don't actually care about the time, just that it's not nil
+				CompletionTime: &metav1.Time{Time: time.Now()},
+			},
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			now := metav1.Now()
+			ctx := config.ToContext(context.Background(), &config.Config{
+				FeatureFlags: &config.FeatureFlags{
+					EnforceNonfalsifiability: config.EnforceNonfalsifiabilityWithSpire,
+				},
+			})
+			if cmp.Diff(c.pod, corev1.Pod{}) == "" {
+				c.pod = corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "pod",
+						Namespace:         "foo",
+						CreationTimestamp: now,
+					},
+					Status: c.podStatus,
+				}
+			}
+
+			startTime := time.Date(2010, 1, 1, 1, 1, 1, 1, time.UTC)
+			tr := v1beta1.TaskRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "task-run",
+					Namespace: "foo",
+				},
+				Status: v1beta1.TaskRunStatus{
+					TaskRunStatusFields: v1beta1.TaskRunStatusFields{
+						StartTime: &metav1.Time{Time: startTime},
+					},
+				},
+			}
+			if c.specifyTaskRunResult {
+				// Specify result
+				tr.Status.TaskSpec = &v1beta1.TaskSpec{
+					Results: []v1beta1.TaskResult{{
+						Name: "some-task-result",
+					}},
+				}
+
+				c.want.TaskSpec = tr.Status.TaskSpec
+			}
+
+			if err := sc.CreateEntries(ctx, &tr, &c.pod, 10000); err != nil {
+				t.Fatalf("unable to create entry for tr: %v", tr.Name)
+			}
+
+			if c.resultOut != nil {
+				id := sc.GetIdentity(&tr)
+				for i := 0; i < 20; i++ {
+					sc.SignIdentities = append(sc.SignIdentities, id)
+				}
+				sigs, err := sc.Sign(ctx, c.resultOut)
+				if err != nil {
+					t.Fatalf("failed to sign: %v", err)
+				}
+				c.resultOut = append(c.resultOut, sigs...)
+				s, err := createMessageFromResults(c.resultOut)
+				if err != nil {
+					t.Fatalf("failed to create message from result: %v", err)
+				}
+
+				c.podStatus.ContainerStatuses[0].State.Terminated.Message = s
+				c.want.TaskRunStatusFields.Steps[0].ContainerState.Terminated.Message = s
+			}
+
+			logger, _ := logging.NewLogger("", "status")
```

Comment:
You probably want https://pkg.go.dev/knative.dev/pkg@v0.0.0-20230612155445-74c4be5e935e/logging/testing#TestLogger - this will add any logs to the test context